### DISCUSSION TITLE
How to set custom AuthoritiesConverter for Jwt tokens?

### DISCUSSION THREAD

**Discussion by sebol** [LogNet/grpc-spring-boot-starter#395] [Q&A]
Hi!

I need to get authorities from a non standard field in Jwt Token, so I need to use Custom Authorites Converter.

I'm trying to do something like this:
`
val providers = mutableListOf<AuthenticationProvider>()

        val jwtAuthenticationConverter = JwtAuthenticationConverter()

        jwtAuthenticationConverter.setJwtGrantedAuthoritiesConverter(MyAuthoritiesConverter())

        val provider = JwtAuthenticationProvider(jwtDecoder)

        provider.setJwtAuthenticationConverter(jwtAuthenticationConverter)

        providers.add(provider)

        return ProviderManager(providers)
`

Generally I see in debug that my authentication provider is registred, but then during the request, default Bearer Authentication provider is choosen instead of mine and default converter is used.

Is there any way to unregister this standard provider or set on it custom converter?

Thanks for help!

**Comment by jvmlet** [comment]
The default one is registered [here GrpcSecurityConfigurerAdapter::configure](https://github.com/LogNet/grpc-spring-boot-starter/blob/master/grpc-spring-boot-starter/src/main/java/org/lognet/springboot/grpc/security/GrpcSecurityConfigurerAdapter.java).
So you can just don;t call the super's `configure` method in your adapter.
